test/hexpm_web/controllers/package_report_controller_test.exs

```diff
@@ -315,11 +318,94 @@ defmodule HexpmWeb.PackageReportControllerTest do
         |> get("/reports/#{report3.id}")
 
       result = response(conn, 200)
-      assert result =~ "id #{report3.id} <\/p>"
-      assert result =~ "on #{report3.package.name} package <\/p>"
-      assert result =~ "<p> #{report3.description} <\/p>"
+      assert result =~ "id #{report3.id}<\/p>"
+      assert result =~ "on #{report3.package.name} package<\/p>"
+      assert result =~ "<p>#{report3.description}<\/p>"
+      # Verify commnets section is not visible
+      refute result =~ "comments-section-div"
+    end
+
+    test "get unresolved for author", %{user3: user3, report5: report5} do
+      conn =
+        build_conn()
+        |> test_login(user3)
+        |> get("/reports/#{report5.id}")
+
+      result = response(conn, 200)
+      assert result =~ "id #{report5.id}<\/p>"
+      assert result =~ "on #{report5.package.name} package<\/p>"
+      assert result =~ "<p>#{report5.description}<\/p>"
+      # Verify commnets section is visible
+      assert result =~ "comments-section-div"
+    end
+
+    test "get unresolved for moderator", %{user2: user2, report5: report5} do
+      conn =
+        build_conn()
+        |> test_login(user2)
+        |> get("/reports/#{report5.id}")
+
+      result = response(conn, 200)
+      assert result =~ "id #{report5.id}<\/p>"
+      assert result =~ "on #{report5.package.name} package<\/p>"
+      assert result =~ "<p>#{report5.description}<\/p>"
+      # Verify commnets section is visible
+      assert result =~ "comments-section-div"
+    end
+
+    test "get unresolved for owner", %{user1: user1, report5: report5} do
+      conn =
+        build_conn()
+        |> test_login(user1)
+        |> get("/reports/#{report5.id}")
+
+      result = response(conn, 200)
+      assert result =~ "id #{report5.id}<\/p>"
+      assert result =~ "on #{report5.package.name} package<\/p>"
+      assert result =~ "<p>#{report5.description}<\/p>"
+      # Verify commnets section is visible
+      assert result =~ "comments-section-div"
+    end
+
+    test "get unresolved for others", %{user4: user4, report5: report5} do
+      conn =
+        build_conn()
+        |> test_login(user4)
+        |> get("/reports/#{report5.id}")
+
+      result = response(conn, 200)
+      assert result =~ "id #{report5.id}<\/p>"
+      assert result =~ "on #{report5.package.name} package<\/p>"
+      assert result =~ "<p>#{report5.description}<\/p>"
       # Verify commnets section is not visible
       refute result =~ "comments-section-div"
     end
   end
+
+  describe "solve/2" do
+    test "get marked package after solve report", %{
+      report2: report2,
+      package1: package1,
+      user2: user2,
+      release1: release1
+    } do
+      conn =
+        build_conn()
+        |> test_login(user2)
+        |> post("/reports/#{report2.id}/solve")
+
+      result = response(conn, 302)
+
+      conn1 =
+        build_conn()
+        |> test_login(user2)
+        |> get("/packages/#{package1.name}")
```

Comment:
Instead of testing the results on a separate page it is enough to fetch the release from the database and checking the retirement on the schema. You can use `Hexpm.Repository.Releases.get` or `Hexpm.Repo.get`.